### DISCUSSION TITLE
Social login with github takes 2 clicks

### DISCUSSION THREAD

**Discussion by JovanShandro** [ory/kratos#1272] [Q&A] [answered]
**Describe the bug**

If the user hasn't already been registered with github, it will take 2 clicks of the 'sign in with github'. After the first one the user data is stored in the database, but the user is still not authenticated. The sign in button of the form has to be pressed one more time for the user to be authenticated.  If the user exists has registered before, it will take him only one click to sign in.  I noticed that the same thing happens in the video tutorial in the link below, so I guess this is intentional, but I wanted to ask is there anything that can be done to change this? 

**Reproducing the bug**
Go through the tutorial here https://www.ory.sh/kratos/docs/guides/sign-in-with-github-google-facebook-linkedin/#github.

**Expected behavior**

The user would be able to authenticate in one click at all times.

**Comment by aeneasr** [comment]
It is one click if the upstream OIDC provider is able to skip login/consent for already authorized applications. What OIDC provider are you using?

**Comment by JovanShandro** [comment]
@aeneasr  Thank you for the quick reply! I have just created an OAuth app on my github account and used the client id and secret from there, but I think I found what is causing it. This is my oidc.github.jsonnet file:
```
    local claims = {
      email_verified: false
    } + std.extVar('claims');

    {
      identity: {
        traits: {
          // Allowing unverified email addresses enables account
          // enumeration attacks, especially if the value is used for
          // e.g. verification or as a password login identifier.
          //
          // Therefore we only return the email if it (a) exists and (b) is marked verified
          // by GitHub.
          [if "email" in claims && claims.email_verified then "email" else null]: claims.email,
        },
      },
    }
```
So my question is:
Is it safe to make the login work with a single click? If yes, how should I change the jsonnet file to do so, or just make the last line like this:
```
[if  claims.email_verified then "email" else null]: claims.email,
```

Thanks again!

**Comment by aeneasr** [comment, marked as answer]
Yes it is safe. Not sure how you should solve this, maybe try it out a bit :)

Converting this to a discussion

**Comment by vinckr** [comment]
Hey @JovanShandro,
just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:

**Comment by trobinpl** [comment]
@JovanShandro Do you have [the session hook configured after the OIDC registration](https://www.ory.sh/kratos/docs/self-service/flows/user-registration#successful-registration)?
